packages/nimble-components/src/rich-text/editor/tests/rich-text-editor.spec.ts

```diff
@@ -642,6 +642,160 @@ describe('RichTextEditor', () => {
                 'P'
             ]);
         });
+
+        describe('Absolute link interactions in the editor', () => {
+            it('should change the text to "nimble-anchor" tag when it is a valid absolute link', async () => {
+                await pageObject.setEditorTextContent(
+                    'https://nimble.ni.dev/ '
```

Comment:
> It is kind of weird that you don't get a link until you've added whitespace after the URL.

We could see this is an expected behavior in every WYSIWYG editor we tried. Here is the [Tiptap issue](https://github.com/ueberdosis/tiptap/issues/3225) where they explicitly enabled the links only when whitespaces are added or by pressing the `Enter` key and not for focusing in and out the editor. Even in other editors like AzDo (only WYSIWYG editor, not markdown editor), enabling links only after they add a whitespace or a new line.

> But if I do `editor.setMarkdown('<http://abc>')`, I can get a rendered link without any whitespace after it.

This is also an expected behavior when we relate it with the markdown parser we use to convert the markdown string to DOM. In this case, when we do `editor.setMarkdown('<http://abc>')`, the string is already parsed into a link according to [Autolink](https://spec.commonmark.org/0.30/#autolinks) in CommonMark flavor. Whereas in the above case, the editor might not know whether the newly typed text is a valid link or not.

We also couldn't find a way to make any configurations to make the editor detect links for every keystroke and it might also create confusion or misinterpretation. For example, when the user wants to type URL `https://nimble.ni.dev`, if it is detected in every keystroke, it will change it to a link after it reaches `https://n`. Also, it becomes difficult for us to modify the `href` of the link based on every update. (As we don't have any solution for now to update `href` based on the `<a>` tag text contents)